### PR TITLE
feat: add config validation warnings

### LINKED ISSUE #3412: config validation: fatal errors vs. warnings
Today all validation results are fatal errors - it prevents Talos from booting, etc.

But we could also report some warnings in addition to fatal errors.

Warning could be printed to the console (kernel logs), but config might validate without fatal errors.

Examples:

* using CNI `custom` with empty manifests should be a warning (use CNI `none` instead #3411 )

### PR DESCRIPTION
**Pull request by AlekSi**
Open questions:
* What should we do if there are validation warnings on the server? Log them?
* Should we return validation warnings in API responses?

Closes #3412.
Refs #3413.

- [x] you linked an issue (if applicable)
- [x] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

### CODE REVIEW

File: cmd/talosctl/cmd/mgmt/validate.go [hunk 1]
```diff
@@ -56,5 +67,6 @@ func init() {
 		fmt.Sprintf("the mode to validate the config for (valid values are %s, %s, and %s)", runtime.ModeMetal.String(), runtime.ModeCloud.String(), runtime.ModeContainer.String()),
 	)
 	cli.Should(validateCmd.MarkFlagRequired("mode"))
+	validateCmd.Flags().BoolVarP(&validateStrictArg, "strict", "", false, "treat validation warnings as errors")
 	addCommand(validateCmd)
 }
```
Comment (AlekSi) [inline cmd/talosctl/cmd/mgmt/validate.go:72]: Do we have tests or scripts where we run `talosctl validate` so this code can be tested?

File: internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go [hunk 2]
```diff
@@ -554,7 +554,9 @@ func receiveConfigViaMaintenanceService(ctx context.Context, logger *log.Logger,
 // ValidateConfig validates the config.
 func ValidateConfig(seq runtime.Sequence, data interface{}) (runtime.TaskExecutionFunc, string) {
 	return func(ctx context.Context, logger *log.Logger, r runtime.Runtime) error {
-		return r.Config().Validate(r.State().Platform().Mode())
+		_, err := r.Config().Validate(r.State().Platform().Mode())
```
Comment (smira) [inline internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go:557]: log.Printf("WARNING: %s")

logger.Printf("WARNING: %s")

````
[talos] validateConfig: WARNING: 
```

File: internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go [hunk 3]
```diff
@@ -539,7 +539,7 @@ func receiveConfigViaMaintenanceService(ctx context.Context, logger *log.Logger,
 		return nil, fmt.Errorf("failed to create config provider: %w", err)
 	}
 
-	if err = provider.Validate(r.State().Platform().Mode()); err != nil {
+	if _, err = provider.Validate(r.State().Platform().Mode()); err != nil {
```
Comment (smira) [inline internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go:542]: log warnings

**Review by smira — COMMENTED** [review]
+ nice idea about exposing warning in apply-config API

File: cmd/talosctl/cmd/talos/edit.go [hunk 4]
```diff
@@ -140,6 +141,11 @@ or 'notepad' for Windows.`,
 
 						continue
 					}
+					for _, m := range resp.GetMessages() {
```
Comment (smira) [inline cmd/talosctl/cmd/talos/edit.go:144]: :+1:

**Review by smira — APPROVED** [review]
:cool:

**Pull request merged**